core/src/test/java/org/apache/iceberg/hadoop/HadoopTableTestBase.java

```diff
@@ -113,17 +113,17 @@ public class HadoopTableTestBase {
           .withRecordCount(2) // needs at least one record or else metrics will filter it out
           .build();
 
-  @Rule public TemporaryFolder temp = new TemporaryFolder();
+  @TempDir Path temp;
```

Comment:
```suggestion
  @TempDir File tableDir;
```
and then we can remove all `this.tableDir = temp.toFile()` calls

Reply:
Thank you for the suggestion. I've modified the code based on your suggestions.